File: app/src/main/java/com/example/geotaskaireminder/dao/TaskDao.java
package com.example.geotaskaireminder.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.geotaskaireminder.model.TaskEntity;

import java.util.List;

@Dao
public interface TaskDao {
    @Query("SELECT * FROM tasks ORDER BY updated_at DESC")
    LiveData<List<TaskEntity>> getAllTasks();

    @Query("SELECT * FROM tasks WHERE id = :taskId LIMIT 1")
    LiveData<TaskEntity> getTaskById(long taskId);

    @Query("SELECT * FROM tasks WHERE id = :taskId LIMIT 1")
    TaskEntity getTaskByIdSync(long taskId);

    @Insert
    long insert(TaskEntity task);

    @Update
    int update(TaskEntity task);

    @Delete
    int delete(TaskEntity task);

    @Query("DELETE FROM tasks WHERE id = :taskId")
    int deleteById(long taskId);

    @Query("UPDATE tasks SET reminder_triggered = :isTriggered, updated_at = :updatedAt WHERE id = :taskId")
    int updateReminderTriggered(long taskId, boolean isTriggered, long updatedAt);
}

File: app/src/main/java/com/dinesh/geotaskai/data/TaskDao.kt
package com.dinesh.geotaskai.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface TaskDao {
    @get:Query("SELECT * FROM tasks ORDER BY updated_at DESC")
    val allTasks: LiveData<List<TaskEntity>>

    @Query("SELECT * FROM tasks WHERE id = :taskId LIMIT 1")
    fun getTaskById(taskId: Long): LiveData<TaskEntity?>

    @Query("SELECT * FROM tasks WHERE id = :taskId LIMIT 1")
    fun getTaskByIdSync(taskId: Long): TaskEntity?

    @Insert
    fun insert(task: TaskEntity): Long

    @Update
    fun update(task: TaskEntity): Int

    @Delete
    fun delete(task: TaskEntity): Int

    @Query("DELETE FROM tasks WHERE id = :taskId")
    fun deleteById(taskId: Long): Int

    @Query("UPDATE tasks SET reminder_triggered = :isTriggered, updated_at = :updatedAt WHERE id = :taskId")
    fun updateReminderTriggered(taskId: Long, isTriggered: Boolean, updatedAt: Long): Int
}
